### PR TITLE
NH-66480: update workflow as amzlinux only work with checkout/action@v3 (or less)

### PR DESCRIPTION
**Pull request by xuan-cao-swi**

### CODE REVIEW

File: .github/workflows/build_for_github_package.yml [hunk 1]
```diff
@@ -40,7 +40,6 @@ jobs:
       - name: Build gem and publish to github package
         id: gemstep
         run:  |
-          mkdir ~/.gem
```
Comment (cheempz) [inline .github/workflows/build_for_github_package.yml:43]: So ~/.gem dir is guaranteed to exist?

**Review by cheempz — DISMISSED** [review]
👍 curiosity question about the removal of mkdir

File: .github/workflows/build_for_github_package.yml (hunk 1 above)
Comment (xuan-cao-swi) [reply .github/workflows/build_for_github_package.yml:43]: Yes, when I first ran it, it indicates the folder already exist, which is different from apm-ruby. I guess the different ruby version make it different. However, I change the workflow to create one if not exist and not create one if exist.

**Review by cheempz — APPROVED** [review]
Thanks @xuan-cao-swi!

**Pull request merged**